### PR TITLE
x-pack/filebeat/input/{cel,httpjson}: clean up tempdir handling in tests

### PR DESCRIPTION
**Pull request by efd6**
## What does this PR do?

Cleans up tempdir handling logic in tests.

## Why is it important?

Clarity/separation of test table from test logic and removal of impinging conflict diff for another PR that is pending.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ] Recommend review with hide whitespace differences.

## Related issues

-

### CODE REVIEW

File: x-pack/filebeat/input/httpjson/input_test.go [hunk 1]
```diff
@@ -329,7 +328,7 @@ func TestInput(t *testing.T) {
 						"value": `[[index .last_response.body "@timestamp"]]`,
 					},
 				},
-				"request.tracer.filename": filepath.Join(tempDirectory, "logs", "http-request-trace-*.ndjson"),
+				"request.tracer.filename": "http-request-trace-*.ndjson",
```
Comment (andrewkroh) [inline x-pack/filebeat/input/httpjson/input_test.go:331]:
```suggestion
				"request.tracer.filename": "logs/http-request-trace-*.ndjson",
```

File: x-pack/filebeat/input/httpjson/input_test.go [hunk 2]
```diff
@@ -1189,6 +1188,12 @@ func TestInput(t *testing.T) {
 			conf := defaultConfig()
 			assert.NoError(t, cfg.Unpack(&conf))
 
+			var tempDir string
+			if conf.Request.Tracer != nil {
+				tempDir = t.TempDir()
+				conf.Request.Tracer.Filename = filepath.Join(tempDir, "logs", conf.Request.Tracer.Filename)
```
Comment (andrewkroh) [inline x-pack/filebeat/input/httpjson/input_test.go:1194]:
```suggestion
				conf.Request.Tracer.Filename = filepath.Join(tempDir, conf.Request.Tracer.Filename)
```

This is to have the "logs" part of the path be specified entirely in the test case.

File: x-pack/filebeat/input/cel/input_test.go [hunk 3]
```diff
@@ -763,38 +761,38 @@ func TestInput(t *testing.T) {
...
+	{
+		name: "tracer_filename_sanitization",
+		server: func(t *testing.T, h http.HandlerFunc, config map[string]interface{}) {
+			server := httptest.NewServer(h)
+			config["resource.url"] = server.URL
+			t.Cleanup(server.Close)
+		},
+		config: map[string]interface{}{
+			"interval":                 1,
+			"resource.tracer.filename": "http-request-trace-*.ndjson",
```
Comment (efd6) [inline x-pack/filebeat/input/cel/input_test.go:791]:
```suggestion
			"resource.tracer.filename": "logs/http-request-trace-*.ndjson",
```

File: x-pack/filebeat/input/cel/input_test.go [hunk 4]
```diff
@@ -1101,6 +1102,12 @@ func TestInput(t *testing.T) {
 				t.Fatalf("unexpected error unpacking config: %v", err)
 			}
 
+			var tempDir string
+			if conf.Resource.Tracer != nil {
+				tempDir = t.TempDir()
+				conf.Resource.Tracer.Filename = filepath.Join(tempDir, "logs", conf.Resource.Tracer.Filename)
```
Comment (efd6) [inline x-pack/filebeat/input/cel/input_test.go:1108]:
```suggestion
				conf.Resource.Tracer.Filename = filepath.Join(tempDir, conf.Resource.Tracer.Filename)
```

**Review by efd6 — COMMENTED** [review]
In CEL for parity

**Review by andrewkroh — APPROVED** [review]

**Pull request merged**